### PR TITLE
Added tests for area actions

### PR DESCRIPTION
**Pull request by sanbornsen**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

### Mandatory
 - [x] What does this PR do? 
This PR adds tests for area actions

- [ ] What issue/task does this PR references?

- [ ] Are the tests Included?
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included

### CODE REVIEW

**Comment by jarifibrahim** [comment]
@sanbornsen I see you've added tests only to `create` actions. Can we have tests that would ensure these actions were `created and dispatched/consumed` ?
Also, I do not see any failure tests. Can we add some failure tests ?

File: karma.conf.js [hunk 1]
```diff
@@ -46,10 +48,12 @@ module.exports = function(config) {
                 'runtime', // explicitly exclude the runtime here
                 'src/app/mock/standalone/sso-api.provider.ts', // this class produces some errors when compiled in test mode
                 'src/app/shared/wit-api.provider.ts', // this class produces some errors when compiled in test mode
-                'src/app/services/login.service.ts' // this requires some dependency from runtime, so exclude it
+                'src/app/services/login.service.ts', // this requires some dependency from runtime, so exclude it
+                'src/tests/*', // excluding files for functional test and page objects
+                'tests/*' // excluding files for functional test and page objects
             ]
         },
-        reporters: ['progress', 'karma-typescript', 'coverage'],
```
[the remark this answers is not in the document]
Comment (sanbornsen) [reply karma.conf.js]: They are different types of reporters. When I switched on the reports to the log it started logging things multiple times for the multiple reporters. I checked and figured, we really need just one or two reporters.

**Comment by sanbornsen** [comment]
> I see you've added tests only to create actions. Can we have tests that would ensure these actions were created and dispatched/consumed ?

@jarifibrahim dispatching actions doesn't happen from here. So writing action dispatcher tests will be written in other places like effects and components. 

> Also, I do not see any failure tests. Can we add some failure tests ?

This is just to test the action classes if they are creating the right instance. Very simple set of tests. Now that you are talking about failure testing, I realise it can be creating an action that requires a payload but while creating we send none. That way it should fail to create an action and throw error. I'll add a test for that.

File: karma.conf.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jarifibrahim) [reply karma.conf.js]: So only `mocha` reporter should be enough?

**Comment by jarifibrahim** [comment]
@sanbornsen I do not see any codecov report for this PR. Please check the CICO logs, it shows `no coverage report found`. Since you've changed the coverage reporter to `mocha`, you'll have to update https://github.com/fabric8-ui/fabric8-planner/blob/master/.codecov.yml file also.

**Comment by sanbornsen** [comment]
> This is just to test the action classes if they are creating the right instance. Very simple set of tests. Now that you are talking about failure testing, I realise it can be creating an action that requires a payload but while creating we send none. That way it should fail to create an action and throw an error. I'll add a test for that.

While writing the test I realised it would fail even before creating an action while compiling. So need no failure test here.

**Comment by jarifibrahim** [comment]
>While writing the test I realised it would fail even before creating an action while compiling. So need no failure test here.

Cool. These tests are pretty simple and we don't really need a failure tests.

File: karma.conf.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanbornsen) [reply karma.conf.js]: Yeah only mocha should be enough

**Pull request merged**